sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs

```diff
@@ -159,11 +139,13 @@ public string IndexName
         /// recordings, instrumentation, etc.
         /// </param>
         /// <returns>A new TestResources context.</returns>
-        public static async Task<SearchResources> CreateWithNoIndexesAsync(SearchTestBase fixture)
+        public static SearchResources CreateWithNoIndexes(SearchTestBase fixture)
         {
-            var resources = new SearchResources(fixture);
-            await resources.CreateSearchServiceAsync();
-            return resources;
+            return new SearchResources(fixture)
+            {
+                // We created no index, but others tests might. We'll check when cleaning up.
+                RequiresCleanup = fixture.Recording.Mode != RecordedTestMode.Playback,
```

Comment:
Why not delete `RequiresCleanup` then and just always do it when not playing back?

Reply:
Because of the shared hotels instance. You created this variable to make sure we don't dispose of it until at the very end. I considered other ways to accomplish this, but figured this was already in place and works, and is no more simple than other ideas I had to know if/when I should clean up.